### PR TITLE
refactor: optimize something

### PR DESCRIPTION
**Pull request by xiyaowong**
see the commit messages for details :)

### CODE REVIEW

**Comment by xiyaowong** [comment]
the main problem is that this will change the buffer, but mapping won't

**Comment by max397574** [comment]
> the main problem is that this will change the buffer, but mapping won't

What exactly?

**Comment by xiyaowong** [comment]
This  is irrelevant to this PR.  It's easy to understand，I don't know how to describe it exactly. Right now, it looks like it's unsolvable. I'm just mentioning it

**Comment by max397574** [comment]
I'd like to merge this into dev branch first
can you change this or open a new pr?

File: lua/better_escape.lua [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
 
-local function start_timeout(timeout)
-  vim.g.better_escape_flag = true
-  local timer = vim.loop.new_timer()
-  timer:start(timeout, 0, function()
-    vim.g.better_escape_flag = false
-  end)
+local function start_timeout()
+  flag = true
+  vim.defer_fn(function()
```
Comment (max397574) [inline lua/better_escape.lua:15]: much better way of doing this

**Comment by max397574** [comment]
thank you very much for the new features and the better way of doing some stuff

**Comment by xiyaowong** [comment]
:)

**Comment by max397574** [comment]
one more thing
the example function would just move the cursor to the right if you are in column <=3 right?

**Comment by xiyaowong** [comment]
this is where i confused, after feeding <bs><bs>, the position(result of `col('.')`) hasn't changed, I tried adding a small amount of delay, but it didn't work

**Comment by max397574** [comment]
ok thx

**Pull request merged**